### ISSUE TITLE
zappa --package doesn't include dist-packages

### ISSUE THREAD

**Issue by monkeypants** [Miserlou/Zappa#905]
`zappa --package` bundles site-packages from the virtualenv. But, on Debian-based distributions, pip installs stuff into `dist-packages`. So, there is a problem that these platforms (e.g. AMI linux) that they don't include pip-installed packages when they `zappa --package`

## Context
Do `zappa --package` on with an AMI linux box, or some other Debian derivative (that does strange things to pip, so it installs stuff into dist-packages).

## Expected Behavior
I would like to see all the stuff from dist-packages in the zappa package, so my thing works the same even if I build it on Debian.

## Actual Behavior
The site-packages is included in the zappa package. The dist-packages are not. The stuff I installed with pip in my virtualenv is not included in the zappa package if I ran it on a Debian derived OS, and my thing doesn't work because it's requirements are not met.

## Possible Fix

 * update create_handler_venv in zappa/core.py so it does stuff with dist-packages for the relevant platforms
 * update detect_flask_apps and detect_django_settings in zappa/utilities.py so they check dist-packages too
 * Is this maybe related to why [/tests/tests.py#L70](https://github.com/Miserlou/Zappa/blob/5641b506090dd3f24e09b0c044c1b47d76f15629/tests/tests.py#L70) is commented out?
 
## Steps to Reproduce
1. Make yourself at home in an AMI linux box
2. Create a virtualenv and activate it.
3. `pip install` a bunch of stuff, such as the requirements for you app
4. See for yourself how there is a bunch of stuff installed in dist-packages
5. do `zappa --package`
6. compare the size of the lambda.zip with one made on non-debian distribution (it's smaller, because it doesn't include the packages from your virtualenv)
7. see how the thing fails with unmet dependencies if you deploy it
8. see how the lambda.zip does not contain your requirements

## Your Environment
* Zappa version used: latest
* Operating System and Python version: Amazon AMI (suspect any debian derivative)
* The output of `pip freeze`: any
* Link to your project (optional): N/A
* Your `zappa_settings.py`: N/A (I think)

**Comment by monkeypants** [comment]
FWIW, this bare-bones app is adequate for reproducing the problem
https://github.com/monkeypants/debug_zappa_905

**Comment by taylor-revcascade** [comment]
Ran into similar problems just trying to stand up a prototype on my locals.  I recommend using a sanitized docker image, something like danielwhatmuff/zappa for building and deploying.  I've baked it into my CI deploys (we use CircleCI) and haven't had an issue since.  I know it's not a fix and ideally it would be platform agnostic, but it's a workaround that I don't feel is too intrusive or counter intuitive to best practices.

**Comment by monkeypants** [comment]
yeah, the  danielwhatmuff/zappa docker image looks like a solution. I'm currently using Jenkins on Ubuntu to work around the problem.

**Comment by dimvoly** [comment]
I too decided to use the Amazon Linux VM ("amzn-ami-hvm-2017.03.1.20170812-x86_64-gp2") to do the "building" of my code since my development laptop is Windows and deploying from Windows using zappa doesn't work at all (understandably). Unfortunately unlike other environments I've tried (like Ubuntu 16.04), Amazon's flavour of Linux puts the pip packages into `dist-packages` instead of `site-packages`. I got around this by copy pasting from one folder to the other for now.

A flow on issue from this is that the `utilities.py` likes to crawl through `dist-packages` and for non utf-8 compatible files throws up errors (pretty harmless errors). Can attest that this happens when parsing "biffh.py" from the package "xlrd".

**Comment by brad-alexander** [comment]
➕ :one: on this, using the new cloud9 IDE from amazon for lambda development, and it sticks everything in dist-packages.  `zappa package` spits out a 40kb file 😭

**Comment by brad-alexander** [comment]
❗️ ❗️ ❗️ 
`unset PYTHON_INSTALL_LAYOUT`
and recreating the virtualenv worked.  https://forums.aws.amazon.com/thread.jspa?messageID=698785

>This is caused by package system-rpm-config installed as part of "@Development tools" group defining PYTHON_INSTALL_LAYOUT=amzon. Without this variable set, everything works fine, because package is installed into site-packages directory which is in search path.

**Comment by mcmonster** [comment]
Amazon-Linux is such a beast sometimes o.O .

I was running into an issue that presented as similar to this issue, though it was related to packages being improperly installed in the system site-packages folder. Amazon-Linux's pip executable is `pip-3.6` while the virtualenv uses `pip3.6`. If you find yourself experiencing maddening deploy issues on Amazon-Linux, check to make sure you are using the correct spelling for pip.

Not really a fix for this ticket, but this is the most similar ticket I came across, so there it is for posterity.

**Comment by dins** [comment]
For me this caused `Unable to import module 'handler': No module named 'werkzeug'` on `zappa update`.
